doc/v5.0/man3/MPI_Win_get_info.3
.\" -*- nroff -*-
.\" Copyright (c) 2015      Research Organization for Information Science
.\"                         and Technology (RIST). All rights reserved.
.\" Copyright (c) 2020      FUJITSU LIMITED.  All rights reserved.
.\" $COPYRIGHT$
.TH MPI_Win_get_info 3 "Sep 30, 2021" "5.0.0rc1" "Open MPI"
.SH NAME
\fBMPI_Win_get_info\fP \- Retrieves active window info hints
.
.SH SYNTAX
.ft R
.SH C Syntax
.nf
#include <mpi.h>
int MPI_Win_get_info(MPI_Win \fIwin\fP, MPI_Info \fI*info_used\fP)
.
.fi
.SH Fortran Syntax
.nf
USE MPI
! or the older form: INCLUDE 'mpif.h'
MPI_WIN_GET_INFO(\fIWIN, INFO_USED, IERROR\fP)
	INTEGER	\fIWIN, INFO_USED, IERROR \fP
.
.fi
.SH Fortran 2008 Syntax
.nf
USE mpi_f08
MPI_Win_get_info(\fIwin\fP, \fIinfo_used\fP, \fIierror\fP)
	TYPE(MPI_Win), INTENT(IN) :: \fIwin\fP
	TYPE(MPI_Info), INTENT(OUT) :: \fIinfo_used\fP
	INTEGER, OPTIONAL, INTENT(OUT) :: \fIierror\fP

.fi
.SH INPUT PARAMETERS
.ft R
.TP 1i
win
Window from which to receive active info hints
.
.SH OUTPUT PARAMETERS
.ft R
.TP 1i
info_used
New info object returned with all active hints on this window.
.TP 1i
IERROR
Fortran only: Error status (integer).
.
.SH DESCRIPTION
.ft R
MPI_Win_get_info returns a new info object containing the hints of
the window associated with
.IR win .
The current setting of all hints actually used by the system related
to this window is returned in
.IR info_used .
If no such hints exist, a handle to a newly created info object is
returned that contains no key/value pair. The user is responsible for
freeing info_used via MPI_Info_free.
.
.SH ERRORS
Almost all MPI routines return an error value; C routines as the value
of the function and Fortran routines in the last argument.
.sp
Before the error value is returned, the current MPI error handler is
called. By default, this error handler aborts the MPI job, except for
I/O function errors. The error handler may be changed with
MPI_Comm_set_errhandler; the predefined error handler
MPI_ERRORS_RETURN may be used to cause error values to be
returned. Note that MPI does not guarantee that an MPI program can
continue past an error.
.
.SH SEE ALSO
MPI_Win_set_info,
MPI_Win_free


doc/v3.0/man3/MPI_Win_get_info.3.php
<?php
$topdir = "../../..";
$title = "MPI_Win_get_info(3) man page (version 3.0.6)";
$meta_desc = "Open MPI v3.0.6 man page: MPI_WIN_GET_INFO(3)";

include_once("$topdir/doc/nav.inc");
include_once("$topdir/includes/header.inc");
?>
<p> <a href="../">&laquo; Return to documentation listing</a></p>
    <!-- manual page source format generated by PolyglotMan v3.2, -->
<!-- available at http://polyglotman.sourceforge.net/ -->

<body bgcolor='white'>
<a href='#toc'>Table of Contents</a><p>

<h2><a name='sect0' href='#toc0'>Name</a></h2>
<b>MPI_Win_get_info</b> - Retrieves active window info hints
<h2><a name='sect1' href='#toc1'>Syntax</a></h2>

<h2><a name='sect2' href='#toc2'>C Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
int MPI_Win_get_info(MPI_Win win, MPI_Info *info_used)
</pre>
<h2><a name='sect3' href='#toc3'>Fortran Syntax</a></h2>
<br>
<pre>USE MPI
! or the older form: INCLUDE &rsquo;mpif.h&rsquo;
MPI_WIN_GET_INFO(WIN, INFO_USED, IERROR)
<tt> </tt>&nbsp;<tt> </tt>&nbsp;INTEGER<tt> </tt>&nbsp;<tt> </tt>&nbsp;COMM, INFO_USED, IERROR
</pre>
<h2><a name='sect4' href='#toc4'>Fortran 2008 Syntax</a></h2>
<br>
<pre>USE mpi_f08
MPI_Win_get_info(win, info_used, ierror)
<tt> </tt>&nbsp;<tt> </tt>&nbsp;TYPE(MPI_Win), INTENT(IN) :: win
<tt> </tt>&nbsp;<tt> </tt>&nbsp;TYPE(MPI_Info), INTENT(OUT) :: info_used
<tt> </tt>&nbsp;<tt> </tt>&nbsp;INTEGER, OPTIONAL, INTENT(OUT) :: ierror
</pre>
<h2><a name='sect5' href='#toc5'>Input Parameters</a></h2>

<dl>

<dt>win </dt>
<dd>Window from which to receive active info hints  </dd>
</dl>

<h2><a name='sect6' href='#toc6'>Output
Parameters</a></h2>

<dl>

<dt>info_used </dt>
<dd>New info object returned with all active hints on this
window. </dd>

<dt>IERROR </dt>
<dd>Fortran only: Error status (integer).  </dd>
</dl>

<h2><a name='sect7' href='#toc7'>Description</a></h2>
MPI_Win_get_info
returns a new info object containing the hints of the window associated
with <i>win</i>. The current setting of all hints actually used by the system related
to this window is returned in <i>info_used</i>. If no such hints exist, a handle
to a newly created info object is returned that contains no key/value pair.
The user is responsible for freeing info_used via <a href="../man3/MPI_Info_free.3.php">MPI_Info_free</a>.
<h2><a name='sect8' href='#toc8'>Errors</a></h2>
Almost
all MPI routines return an error value; C routines as the value of the
function and Fortran routines in the last argument. <p>
Before the error value
is returned, the current MPI error handler is called. By default, this error
handler aborts the MPI job, except for I/O function errors. The error handler
may be changed with <a href="../man3/MPI_Comm_set_errhandler.3.php">MPI_Comm_set_errhandler</a>; the predefined error handler
MPI_ERRORS_RETURN may be used to cause error values to be returned. Note
that MPI does not guarantee that an MPI program can continue past an error.

<h2><a name='sect9' href='#toc9'>See Also</a></h2>
<a href="../man3/MPI_Win_set_info.3.php">MPI_Win_set_info</a>, <a href="../man3/MPI_Win_free.3.php">MPI_Win_free</a> <p>

<hr><p>
<a name='toc'><b>Table of Contents</b></a><p>
<ul>
<li><a name='toc0' href='#sect0'>Name</a></li>
<li><a name='toc1' href='#sect1'>Syntax</a></li>
<li><a name='toc2' href='#sect2'>C Syntax</a></li>
<li><a name='toc3' href='#sect3'>Fortran Syntax</a></li>
<li><a name='toc4' href='#sect4'>Fortran 2008 Syntax</a></li>
<li><a name='toc5' href='#sect5'>Input Parameters</a></li>
<li><a name='toc6' href='#sect6'>Output Parameters</a></li>
<li><a name='toc7' href='#sect7'>Description</a></li>
<li><a name='toc8' href='#sect8'>Errors</a></li>
<li><a name='toc9' href='#sect9'>See Also</a></li>
</ul>


<p> <a href="../">&laquo; Return to documentation listing</a></p>
<?php
include_once("$topdir/includes/footer.inc");
